src/fexecve.c

```diff
@@ -16,7 +16,7 @@ array_of_value(value v)
   size = Wosize_val(v);
   arr = caml_stat_alloc((size + 1) * sizeof(char *));
   for (i = 0; i < size; i++)
-    arr[i] = String_val(Field(v, i));
+    arr[i] = caml_stat_strdup(String_val(Field(v, i)));
```

Comment:
Because of this warning: a `const` string pointer is written into the non-const array.
```
dune build @install
         gcc src/fexecve.o
fexecve.c: In function ‘array_of_value’:
fexecve.c:19:12: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
     arr[i] = String_val(Field(v, i));
            ^
```
However you're right, the copy isn't necessary and I believe a cast removing the const is enough to remove the warning, as the string isn't written into.